rewrite-core/src/main/java/org/openrewrite/scheduling/WatchableExecutionContext.java

```diff
@@ -40,7 +40,7 @@ public void resetHasNewMessages() {
 
     @Override
     public void putMessage(String key, Object value) {
-        hasNewMessages = true;
+        hasNewMessages = !ExecutionContext.CURRENT_RECIPE.equals(key);
```

Comment:
Alternatively, we can reset hasNewMessages in the RecipeScheduler [here](https://github.com/openrewrite/rewrite/blob/main/rewrite-core/src/main/java/org/openrewrite/RecipeScheduler.java#L176):

```java
ctx.putCurrentRecipe(recipe);
if (ctx instanceof WatchableExecutionContext) {
    ((WatchableExecutionContext) ctx).resetHasNewMessages();
}
```